pkg/services/datasources/cache.go

```diff
@@ -33,22 +33,22 @@ func (dc *CacheServiceImpl) Init() error {
 
 func (dc *CacheServiceImpl) GetDatasource(
 	datasourceID int64,
-	user *models.SignedInUser,
+	orgID int64,
 	skipCache bool,
 ) (*models.DataSource, error) {
 	cacheKey := fmt.Sprintf("ds-%d", datasourceID)
 
 	if !skipCache {
 		if cached, found := dc.CacheService.Get(cacheKey); found {
 			ds := cached.(*models.DataSource)
-			if ds.OrgId == user.OrgId {
+			if ds.OrgId == orgID {
 				return ds, nil
 			}
 		}
 	}
 
-	plog.Debug("Querying for data source via SQL store", "id", datasourceID, "orgId", user.OrgId)
-	ds, err := dc.SQLStore.GetDataSourceByID(datasourceID, user.OrgId)
+	plog.Debug("Querying for data source via SQL store", "id", datasourceID, "orgId", orgID)
+	ds, err := dc.SQLStore.GetDataSourceByID(datasourceID, orgID)
```

Comment:
Best to error if orgID is zero, because then there will be no orgid filter on the query.

Reply:
I think that in that case it will fail anyway since it falls to [this](https://github.com/grafana/grafana/blob/ab078133bfe9771f06bd21712f7ad6f9113c8a61/pkg/services/sqlstore/datasource.go#L38) case.

Reply:
It will pass. If the `OrgID` is zero, since the Type of the property is an int64, and the zero value 0, then xorm's engine.Get will not add the 'WHERE org_id=` filter to the query, so it will fetch based on the data source id alone.